pkg/api/types.go

```diff
@@ -302,6 +302,11 @@ type VolumeSource struct {
 	// ScaleIO represents a ScaleIO persistent volume attached and mounted on Kubernetes nodes.
 	// +optional
 	ScaleIO *ScaleIOVolumeSource
+	// LocalStorage represents a local volume accessible only from one node
+	// This volume type cannot be used as a direct volume source and is only required here
```

Comment:
Just checked with Saad.  ConstructVolumeSpec can actually support PersistentVolumeSource too, so then I can remove this from the Volume Source.  :-D